janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphOperationCountingTest.java

```diff
@@ -643,4 +646,25 @@ private double testAllVertices(long vid, int numV) {
         return ((System.nanoTime()-start)/1000000.0);
     }
 
+
+    @Test
+    public void getEdgeByHasId() {
```

Comment:
I just executed this test with the Thrift and also the Berkeley backend and it passed even when I removed your changes to `JanusGraphStep`. I guess we first need to find a way to reproduce the problem with a test case. Although, I can't quite see why this suddenly works in the test case, but didn't when I tried it from the console when I created the issue.

Reply:
For reproduce change 659 and 660 lines to use `eid.toString()`

Reply:
Ok, but why should it work when you provide a string instead of a `RelationIdentifier`? That's not really a supported feature and also not what the issue was about.

Reply:
Yes, right, this is another case related with https://github.com/JanusGraph/janusgraph/pull/1343#issuecomment-447857246

May be `.profile()` will help